### PR TITLE
Allow client to specify fields for v1/projects results

### PR DESCRIPTION
**Pull request by cpeel**
Allow an API client to limit which fields are returned for project searches using the `v1/projects` endpoint. This is in preparation to support a PG use-case.

Testable with curl:
```bash
curl -i -X GET -H "Accept: application/json" -H "X-API-KEY: $API_KEY" \
    "https://www.pgdp.org/~cpeel/c.branch/limit-fields/api/?url=v1/projects&fields=projectid,title,author"
```

### CODE REVIEW

File: api/USERS_GUIDE.md
Comment (chrismiceli) [inline api/USERS_GUIDE.md:133]: Based on the last MR about multi fields, should we instead do field[]=title&field[]=author etc.?

**Review by chrismiceli — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

File: api/v1_projects.inc
Comment (70ray) [inline api/v1_projects.inc:108]: Not really relevant to this MR but I came across the null coalescing operator recently which does exactly the same as array_get so we could write `$return_fields = $query_params["fields"] ?? null;` in native php.

**Review by 70ray — APPROVED** [review]

File: api/USERS_GUIDE.md
Comment (cpeel) [reply api/USERS_GUIDE.md:133]: I get where you're going with that, but I think that will be confusing with this endpoint in particular because we aren't `OR`ing the fields together in any way and they aren't part of the search criteria, for instance:
```
genre[]=Science&genre[]=Sports&field[]=title&field[]=author
```
Thoughts?

Comment (chrismiceli) [reply api/USERS_GUIDE.md:133]: I see where you are coming from. I think it would make sense to a user of the API if we just explained that all fields that accept multiple values are sent with field[] syntax. Then we need to clarify that multivalued search fields are ORed while distinct values that are relevant to search are ANDed. I think that is less confusing then figuring out how to send multi values depending on how they are used by the API.

In reality though, I've seen it done both ways and am open to either.

Comment (cpeel) [reply api/USERS_GUIDE.md:133]: That makes a lot of sense. The comma approach is nice but we can only use it for strings that might not contain commas (like a field name). I agree it makes more sense to be consistent within the API and use `[]` for multiple values. Let me get the branch changed to use that instead. Thanks for bringing it up and talking through it.

**Comment by cpeel** [comment]
Updated syntax for testing:
```bash
curl -i -X GET -H "Accept: application/json" -H "X-API-KEY: $API_KEY" \
    "https://www.pgdp.org/~cpeel/c.branch/limit-fields/api/?url=v1/projects&field[]=projectid&field[]=title&field[]=author"
```

**Comment by chrismiceli** [comment]
I can't find the approve button after your last commit but consider this an approval.

**Review by srjfoo — APPROVED** [review]

File: api/USERS_GUIDE.md
Comment (cpeel) [reply api/USERS_GUIDE.md:151]: I don't think that's necessary as the prior examples demonstrate asking for the full project. I included it in the YAML since there aren't examples there to demonstrate with.

**Pull request merged**
